deploy/01_deploy_vault_factory.ts

```diff
@@ -0,0 +1,32 @@
+import { DeployFunction } from 'hardhat-deploy/types';
+import { THardhatRuntimeEnvironmentExtended } from '../helpers/types/THardhatRuntimeEnvironmentExtended';
+import { ethers, upgrades } from 'hardhat';
+import { VaultFactory } from "../typechain";
+
+const func: DeployFunction = async function (hre: THardhatRuntimeEnvironmentExtended) {
+  const { getNamedAccounts, deployments } = hre;
+  const { deploy } = deployments;
+  const { deployer } = await getNamedAccounts();
+
+  /// ===== VaultTemplate =====
+  const vaultTemplate = await deploy('AssetVault', {
+    from: deployer,
+    log: true,
+  });
+  console.log("Using vaultTemplate deployed at: ", vaultTemplate.address)
+
+  const whitelist = await ethers.getContract("CallWhitelist", deployer);
```

Comment:
It kind of feels like `CallWhitelist` should just be deployed in this file

Reply:
Yup, I agree here. Will move it. Thanks